### PR TITLE
Simlify `ErrorStack` circuit with `ResponsibleOpcode` lookup

### PR DESCRIPTION
**Pull request by silathdiir**
Close https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1154

### Summary

1. Update `valid_stack_ptr_range` function to `invalid_stack_ptrs`. It returns the invalid stack pointers of specified opcode.

2. Add Enum `ResponsibleOp` and update `ExecutionState::responsible_opcodes` to return a vector of ResponsibleOp.

3. Update `FixedTableTag.ResponsibleOpcode` build for invalid stack pointers, and delete `FixedTableTag.OpcodeStack`.

4. Update `ErrorStack` circuit with `ResponsibleOpcode` lookup.

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/step.rs [hunk 1]
```diff
@@ -318,6 +332,31 @@ impl ExecutionState {
...
 }
 
+/// Enum of Responsible opcode mapping to execution state.
+#[derive(Debug)]
+pub(crate) enum ResponsibleOp {
+    /// Raw opcode
+    Op(OpcodeId),
+    /// Corresponding to ExecutionState::ErrorStack
+    InvalidStackPtr(OpcodeId, u32),
+}
```
Comment (han0110) [inline zkevm-circuits/src/evm_circuit/step.rs:342]: Nice!

**Review by han0110 — APPROVED** [review]
LGTM! Very clean refactor!

**Comment by DreamWuGit** [comment]
LGTM,  just want to confirm about cost of new lookup. the old fix lookup is one opcode in one row(opcode, min_stack, max_stack), the new lookup table will be multiple rows for one op code. rows number should be = 0...min_stack + max_stack..1025. @han0110

**Comment by han0110** [comment]
> just want to confirm about cost of new lookup

You mean the row cost on the fixed table? If so I think it should be negligible (compare to the bitwise ones).

**Review by DreamWuGit — APPROVED** [review]
LGTM

**Pull request merged**